brave/src/main/java/brave/internal/recorder/PendingSpan.java

```diff
@@ -15,17 +15,32 @@
 
 import brave.Clock;
 import brave.handler.MutableSpan;
+import brave.internal.Nullable;
+import brave.propagation.TraceContext;
+import java.lang.ref.WeakReference;
 
-public final class PendingSpan {
+/**
+ * This includes a weak reference of the trace context, which allows externalized forms of the trace
+ * context to be swapped for the one in use.
+ *
+ * <p>This is a weak reference to ensure that {@link PendingSpans} can clean up on GC.
+ */
+public final class PendingSpan extends WeakReference<TraceContext> {
```

Comment:
I don't think this type is exposed to users and can be leaked by them. Can't we use a normal reference?

Reply:
if we use a hard reference, it breaks PendingSpans. The keys used in PendingSpans include a weak reference to the trace context, if the value of the same entry has a hard reference, the weak keys are made useless as they can never be collected. Is there a good way to rephrase this?

Reply:
I'll rephrase as it isn't clear that this type is a map entry value anyway..

Reply:
PTAL! thanks